mixins/imageHelper.ts

```diff
@@ -7,106 +7,178 @@ import Vue from 'vue';
+          throw new TypeError(
+            'item.ImageTags must not be null or undefined when an item object is passed'
+          );
+        } else if (!tag) {
+          tag = item.ImageTags[type];
         }
-      } else if (item.Type === 'Episode' && type === ImageType.Backdrop) {
-        itemId = item.SeriesId;
-      } else if (item.Type === 'Audio' && type === ImageType.Backdrop) {
-        itemId = item.AlbumArtists?.[0].Id;
-      } else {
-        itemId = item.Id;
+      } else if (!itemId) {
+        throw new TypeError(
+          'itemId must not be null or undefined when an item object is not passed'
+        );
       }
 
       const url = new URL(
         `${this.$axios.defaults.baseURL}/Items/${itemId}/Images/${type}`
       );
 
-      let imageTag;
-      if (item.Type === 'Episode' && type === ImageType.Thumb) {
-        if (item.SeriesThumbImageTag) {
-          imageTag = item.SeriesThumbImageTag;
-        } else {
-          imageTag = item.ParentBackdropImageTags?.[0];
-        }
-      } else if (item.Type === 'Episode' && type === ImageType.Backdrop) {
-        imageTag = item.ParentBackdropImageTags?.[0];
-      } else {
-        imageTag = item.ImageTags?.[type];
-      }
-
       const params: { [k: string]: string | number | undefined } = {
-        tag: imageTag,
-        quality: 90
+        tag,
+        quality
       };
 
-      if (element && limitByWidth) {
-        params.maxWidth = element.clientWidth.toString();
-      } else if (element) {
-        params.maxHeight = element.clientHeight.toString();
+      if (limitByWidth && maxWidth) {
+        params.maxWidth = maxWidth.toString();
+      } else if (maxHeight) {
+        params.maxHeight = maxHeight.toString();
       }
 
       url.search = stringify(params);
 
       return url.toString();
+    },
+    /**
+     * Shorthand for the 'getImageUrlForElement' function when using an itemId only.
+     *
+     * @param {ImageType} type - The type of the image to fetch.
+     * @param {string} id - iitemId to get the image from.
+     * @returns {string} The URL for the image, with the base URL set and the options provided.
+     */
+    getImageUrlById(type: ImageType, id: string): string {
+      return this.getImageUrlForElement({
+        type,
+        itemId: id
+      });
+    },
+    /**
+     * Shorthand for the 'getImageUrlForElement' function when using an itemId and a DOM element for size calculations.
+     *
+     * @param {ImageType} type - The type of the image to fetch.
+     * @param {string} id - itemId to get the image from.
+     * @param {HTMLElement} element - The DOM element which size will be used for the image's maximum width or height (optional).
+     * @returns {string} The URL for the image, with the base URL set and the options provided.
+     */
+    getSizedImageUrlById(
+      type: ImageType,
+      id: string,
+      element: HTMLElement
+    ): string {
+      return this.getImageUrlForElement({
```

Comment:
Same for this, please just call the original one instead of making multiple aliases that all go to the same thing.